code/game/turfs/open/floor/plating/thematic.dm

```diff
@@ -0,0 +1,12 @@
+/turf/thematic
+	var/trait
+
+/turf/thematic/New()
```

Comment:
You can't expect to change ~255*255 (excluding the lava/station turfs) turfs to another theme and not expect major lag. Especially if its on New() (Why is it on new in the first place)

Reply:
mining literally already did this when minerals floyddabbed on New and changed their turf to whatever mineral holds the ore they generated. gobless thats gone